src/ray/gcs/gcs_server/test/gcs_placement_group_manager_test.cc

```diff
@@ -431,43 +431,40 @@ TEST_F(GcsPlacementGroupManagerTest, TestRescheduleWhenNodeDead) {
   RegisterPlacementGroup(request1, [&registered_placement_group_count](Status status) {
     ++registered_placement_group_count;
   });
-  auto request2 = Mocker::GenCreatePlacementGroupRequest();
-  RegisterPlacementGroup(request2, [&registered_placement_group_count](Status status) {
-    ++registered_placement_group_count;
-  });
-  ASSERT_EQ(registered_placement_group_count, 2);
+  ASSERT_EQ(registered_placement_group_count, 1);
   WaitForExpectedPgCount(1);
   auto placement_group = mock_placement_group_scheduler_->placement_groups_.back();
-  placement_group->GetMutableBundle(0)->set_node_id(NodeID::FromRandom().Binary());
-  placement_group->GetMutableBundle(1)->set_node_id(NodeID::FromRandom().Binary());
   mock_placement_group_scheduler_->placement_groups_.pop_back();
+  OnPlacementGroupCreationSuccess(placement_group);
+
   // If a node dies, we will set the bundles above it to be unplaced and reschedule the
-  // placement group. The placement group state is set to `RESCHEDULING` and will be
-  // scheduled first.
+  // placement group. The placement group state is set to `RESCHEDULING`
   mock_placement_group_scheduler_->group_on_dead_node_ =
       placement_group->GetPlacementGroupID();
   mock_placement_group_scheduler_->bundles_on_dead_node_.push_back(0);
   gcs_placement_group_manager_->OnNodeDead(NodeID::FromRandom());
```

Comment:
This part kind of doesn't make sense. How is killing a random node cause rescheduling? Isn't it supposed to be a node that pg is scheduled on?

Reply:
This is actually where the original code was not written well enough. Its mock is written like this, and it is difficult for me to explain it. I think this test case is simply fixed for now. Then it is better to refactor this gcs_placement_group_manager_test in a new way in the future.